src/main/app/claims/converters/claimantResponseConverter.ts

```diff
@@ -37,27 +39,53 @@ export class ClaimantResponseConverter {
-        rejectionReason: courtDetermination.rejectionReason ?
-          courtDetermination.rejectionReason.text : undefined,
-        disposableIncome: courtDetermination.disposableIncome ?
-          courtDetermination.disposableIncome : 0,
-        decisionType: courtDetermination.decisionType
-      }
+      respAcceptance.courtDetermination = courtDetermination
     }
-    const claimantPaymentIntention: ModelPaymentIntention = draftClaimantResponse.alternatePaymentMethod
+    const claimantPaymentIntention: DomainPaymentIntention = this.getClaimantPaymentIntention(draftClaimantResponse)
     if (claimantPaymentIntention) {
-      respAcceptance.claimantPaymentIntention = claimantPaymentIntention.toDomainInstance()
+      respAcceptance.claimantPaymentIntention = claimantPaymentIntention
+    }
+    return respAcceptance
+  }
+
+  private static getCourtDetermination (courtDetermination: CourtDetermination): DomainCourtDetermination {
+    const decisionType: DecisionType = courtDetermination.decisionType
+    if (decisionType === DecisionType.COURT && !courtDetermination.courtPaymentIntention) {
+      throw new Error('court offered payment intention not found where decision type is COURT')
+    }
+    if (decisionType === DecisionType.CLAIMANT_IN_FAVOUR_OF_DEFENDANT) {
+      return undefined
+    }
+    if (!courtDetermination.courtPaymentIntention && !courtDetermination.courtDecision) {
+      return undefined
+    }
+    const responseCourtDetermination: DomainCourtDetermination = new DomainCourtDetermination()
+    responseCourtDetermination.courtDecision = courtDetermination.courtDecision
+    responseCourtDetermination.courtPaymentIntention = courtDetermination.courtPaymentIntention
+    if (courtDetermination.rejectionReason) {
+      responseCourtDetermination.rejectionReason = courtDetermination.rejectionReason.text
+    }
+    responseCourtDetermination.disposableIncome = courtDetermination.disposableIncome ? courtDetermination.disposableIncome : 0
+    responseCourtDetermination.decisionType = courtDetermination.decisionType
+    return responseCourtDetermination
+  }
 
-      if (claimantPaymentIntention.paymentOption.option.value === PaymentOption.IMMEDIATELY) {
-        respAcceptance.claimantPaymentIntention.paymentDate = MomentFactory.currentDate().add(5, 'days')
+  private static getClaimantPaymentIntention (draftClaimantResponse: DraftClaimantResponse): DomainPaymentIntention {
+    const decisionType = draftClaimantResponse.courtDetermination.decisionType
+    if (draftClaimantResponse.alternatePaymentMethod) {
+      const claimantPaymentIntention = draftClaimantResponse.alternatePaymentMethod.toDomainInstance()
+      if (draftClaimantResponse.alternatePaymentMethod.paymentOption.option.value === PaymentOption.IMMEDIATELY) {
+        claimantPaymentIntention.paymentDate = MomentFactory.currentDate().add(5, 'days')
+      }
+      return claimantPaymentIntention
+    } else {
```

Comment:
this should just be an } else if { } rather than an else { if () }

Reply:
done